### PR TITLE
[V2] Update v2

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist
- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

- Updated tasks and their metadata  
- Updated the score function in `AbsTaskRetrieval`

### CODE REVIEW

**Comment by orionw** [comment]
Thanks @Samoed!  Score function looks good. Is the ColBERT stuff new here - I am a bit confused by the commit threads.

**Comment by Samoed** [comment]
Yes, this is a new model and was integrated into the main branch, but GitHub is displaying it a bit oddly. I've added you to review the `score_function` changes in `RetrievalEvaluator`. If the `RetrievalEvaluator` changes look good, I’ll proceed with merging into `v2`.

**Comment by orionw** [comment]
Yes refactor looks good! Only one small nit, model.score in the model classes as a replacement is a bit ambiguous, would prefer something like score_fn or distance_fn or something more verbose.

Otherwise LGTM, feel free to merge.

File: mteb/evaluation/evaluators/model_classes.py [hunk 1]
```diff
@@ -338,17 +320,20 @@ def _full_corpus_search(
             logging.info("Computing Similarities...")
             query_embeddings = torch.as_tensor(query_embeddings).to(device)
             sub_corpus_embeddings = torch.as_tensor(sub_corpus_embeddings).to(device)
+
+            score_function = (
+                self.model.score if hasattr(self.model, "score") else cos_sim
```
Comment (orionw) [inline mteb/evaluation/evaluators/model_classes.py:325]: This was the line that confused me, but I am only on mobile today so perhaps I’m misreading

Comment (Samoed) [reply mteb/evaluation/evaluators/model_classes.py:325]: You are right. This should be `similarity` function as in sentence_transformers

**Pull request merged**